tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -1682,6 +1720,47 @@ class SpirvVectorShuffle : public SpirvInstruction {
   llvm::SmallVector<uint32_t, 4> components;
 };
 
+class SpirvArrayLength : public SpirvInstruction {
+public:
+  SpirvArrayLength(QualType resultType, uint32_t resultId, SourceLocation loc,
+                   SpirvInstruction *structure, uint32_t arrayMember);
+
+  // For LLVM-style RTTI
+  static bool classof(const SpirvInstruction *inst) {
+    return inst->getKind() == IK_ArrayLength;
+  }
+
+  DECLARE_INVOKE_VISITOR_FOR_CLASS(SpirvArrayLength)
+
+  SpirvInstruction *getStructure() const { return structure; }
+  uint32_t getArrayMember() const { return arrayMember; }
+
+private:
+  SpirvInstruction *structure;
+  uint32_t arrayMember;
+};
+
+class SpirvLineInfo : public SpirvInstruction {
```

Comment:
Do we want to keep an in-memory representation for debug line info? Better to leave debug info attached to whatever entity they are annotating and have a pass write out the binaries for them directly. Probably add a TODO to clean up later?